File: 38-count-and-say/count-and-say.java
class Solution {
    public String countAndSay(int n) {
        String answer = "1";
        for (; n > 1; n--) {
            StringBuilder sb = new StringBuilder();
            char prev = answer.charAt(answer.length()-1);
            int count = 1;
            for (int i=answer.length()-2; i>=0; i--) {
                char cur = answer.charAt(i);
                if (cur == prev) count++;
                else {
                    sb.insert(0, prev);
                    sb.insert(0, count);
                    prev = cur;
                    count = 1;
                }
            }
            sb.insert(0, prev);
            sb.insert(0, count);
            answer = sb.toString();
        }
        return answer;
    }
}

File: 38-count-and-say/count-and-say.py
# 9:36

class Solution:
    def countAndSay(self, n: int) -> str:
        nums = [1]
        for _ in range(n-1):
            next_nums = []
            prev_num = nums[0]
            count = 1
            for i in range(1, len(nums)):
                num = nums[i]
                if num == prev_num:
                    count += 1
                else:
                    next_nums.append(count)
                    next_nums.append(prev_num)
                    prev_num = num
                    count = 1
            next_nums.append(count)
            next_nums.append(prev_num)
            nums = next_nums            
        return "".join([str(num) for num in nums])